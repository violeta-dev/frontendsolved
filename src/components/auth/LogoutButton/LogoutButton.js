import React from 'react';
import T from 'prop-types';
import { LogoutOutlined } from '@ant-design/icons';

import ConfirmationButton from '../../shared/ConfirmationButton';

const LogoutButton = ({ onLogout, ...props }) => (
  <ConfirmationButton
    danger
    icon={<LogoutOutlined />}
    shape="round"
    type="dashed"
    confirmationProps={{
      title: 'Close session?',
      content: 'Are you sure you want to disconnect?',
      okText: 'Yes',
      cancelText: 'No',
      okButtonProps: {
        danger: true,
      },
    }}
    onConfirm={onLogout}
    {...props}
  />
);

LogoutButton.propTypes = {
  onLogout: T.func.isRequired,
};

export default LogoutButton;
