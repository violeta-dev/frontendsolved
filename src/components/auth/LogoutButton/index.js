import { connect } from 'react-redux';

import LogoutButton from './LogoutButton';
import { logout } from '../../../store/actions';

const mapDispatchToProps = {
  onLogout: logout,
};

export default connect(null, mapDispatchToProps)(LogoutButton);
