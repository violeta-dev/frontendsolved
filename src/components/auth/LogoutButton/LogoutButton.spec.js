import React from 'react';
import { shallow } from 'enzyme';

import LogoutButton from './LogoutButton';

describe('LogoutButton', () => {
  const props = { onLogout: jest.fn() };
  const render = () => shallow(<LogoutButton {...props} />);
  test('snapshot', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });

  test('should have round shape', () => {
    const wrapper = render();
    expect(wrapper.props().shape).toBe('round');
  });

  test('should have type dashed', () => {
    const wrapper = render();
    expect(wrapper.props().type).toBe('dashed');
  });

  test('should call onLogout', () => {
    const wrapper = render();
    wrapper.props().onConfirm();
    expect(props.onLogout).toHaveBeenCalled();
  });
});
