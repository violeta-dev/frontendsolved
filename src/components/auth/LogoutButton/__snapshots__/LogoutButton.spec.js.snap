// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton snapshot 1`] = `
<ConfirmationButton
  confirmationProps={
    Object {
      "cancelText": "No",
      "content": "Are you sure you want to disconnect?",
      "okButtonProps": Object {
        "danger": true,
      },
      "okText": "Yes",
      "title": "Close session?",
    }
  }
  danger={true}
  icon={<ForwardRef(LogoutOutlined) />}
  onConfirm={[MockFunction]}
  shape="round"
  type="dashed"
/>
`;
